controller_interface/test/test_controller_with_options.hpp

```diff
@@ -31,11 +34,24 @@ class ControllerWithOptions : public controller_interface::ControllerInterface
 {
 public:
   ControllerWithOptions() = default;
+  controller_interface::return_type on_init(const std::string & controller_name) override
+  {
+    std::cout << controller_name << std::endl;   // temporary line to remove warning
+    lifecycle_state_ = rclcpp_lifecycle::State(
+      lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED,
+      controller_interface::state_names::UNCONFIGURED);
+    return controller_interface::return_type::OK;
+  }
+
   controller_interface::return_type init(const std::string & controller_name) override
   {
     rclcpp::NodeOptions options;
     options.allow_undeclared_parameters(true).automatically_declare_parameters_from_overrides(true);
-    auto result = ControllerInterface::init(controller_name, options);
+    node_ = std::make_shared<rclcpp::Node>(
+      controller_name,
+      options.allow_undeclared_parameters(true));
```

Comment:
already done in line 49 so it can be written shorter.

```suggestion
    node_ = std::make_shared<rclcpp::Node>(controller_name, options);
```